assets/js/googlesitekit/api/index.test.js

```diff
@@ -436,6 +479,47 @@ describe( 'googlesitekit.api', () => {
 			);
 			expect( cacheData.cacheHit ).toEqual( false );
 		} );
+
+		it( 'should add tracking info to the data layer when an error is returned on set', async () => {
+			let pushArgs;
+			const dataLayer = {
+				[ DATA_LAYER ]: {
+					push: ( ...args ) => pushArgs = args,
+				},
+			};
+			const config = {
+				trackingEnabled: true,
+			};
+			const { trackEvent } = createTracking( config, dataLayer );
+
+			// Replace the trackEvent implementation to use our version with the mocked dataLayer.
+			const trackEventSpy = jest.spyOn( Tracking, 'trackEvent' ).mockImplementation( trackEvent );
+
+			const errorResponse = {
+				code: 'internal_server_error',
+				message: 'Internal server error',
+				data: { status: 500 },
+			};
+
+			fetchMock.postOnce(
+				/^\/google-site-kit\/v1\/core\/search-console\/data\/settings/,
+				{ body: errorResponse, status: 500 }
+			);
+
+			try {
+				await set( 'core', 'search-console', 'settings', 'data' );
```

Comment:
Other comments above about using test values for the arguments apply here as well 🙂

Reply:
👍

Reply:
These are still yet to be updated 😄

Reply:
Fixed